node-tetra/src/it/scala/co/topl/tetra/it/util/NodeDockerApi.scala

```
@@ -0,0 +1,44 @@
+package co.topl.tetra.it.util
+
+import com.spotify.docker.client.DockerClient
+import org.slf4j.{Logger, LoggerFactory}
+
+case class NodeDockerApi(containerId: String)(implicit dockerClient: DockerClient) {
+
+  private val timeout = 10
+  val logger: Logger = LoggerFactory.getLogger(this.toString)
+
+  def start(): Unit = {
+    logger.info(s"Starting")
+    dockerClient.startContainer(containerId)
+    awaitContainerStart()
+    logger.info(s"Successfully started container $containerId on IP ${ipAddress()}")
+  }
+
+  def stop(): Unit = {
+    logger.info(s"Stopping")
+    dockerClient.stopContainer(containerId, timeout)
+  }
+
+  def ipAddress(): String =
+    dockerClient.inspectContainer(containerId).networkSettings().ipAddress()
+
+  private def awaitContainerStart(): Unit = {
```

maybe we can avoid throwing exceptions, and returning options or Eithers, this function sometimes returns Unit, and sometimes exceptions

---

we could throw an exception if we don't want to do any recovery at all, and I believe it is our case.